scripts/legislation/africa/ghana_english.sh

```diff
@@ -0,0 +1,215 @@
+#!/bin/bash
+
+function remove_all_text_before_first_header {
+  sed -n '/^PART ONE2/,$p'
+}
+
+function remove_footnotes {
+  sed '/^I\. This Act is consolidated/d' | \
+    sed '/^2\. Enacted as the/d'
+}
+
+function prefix_article_numbers_with_article_literal {
+  sed -E 's/^([0-9]+\.)/Article \1/'
+}
+
+function append_pipe_to_article_title {
+  sed -E 's/^(Article .*)$/\1|/'
```

Comment:
Could you explain what this function does?

Reply:
Certainly. This function works in concert with `prefix_article_numbers_with_article_literal` and `move_article_titles_above_article_bodies`.

You may recall that in the unprocessed legislation, the lines containing article numbers consist of the article number followed by a period followed by the article title. The `prefix_article_numbers_with_article_literal` function replaces each article number - period combination with the word 'Article' followed by the article number. The `append_pipe_to_article_title` function then adds the pipe character to the end of the article title. This script invokes these two functions before the `apply_common_transformations_to_stdin` function, which, among other things, replaces each 'Article' - number combination with the number in parentheses and replaces all newline characters with spaces. In the absence of newline characters, the pipe characters added by the `append_pipe_to_article_title` function separate article titles from article bodies. The `move_article_titles_above_article_bodies` function then relies on these pipe characters when identifying the text to move. I selected the pipe character as the delimiter because I wanted to use a character that does not otherwise appear in the body of the legislation.

Reply:
Thanks for the explanation.